ui/perfherder/compare/CompareTable.jsx

```diff
@@ -79,6 +91,58 @@ export default class CompareTable extends React.PureComponent {
     }
   };
 
+  // Sorts the specified key of the array data depending on the typeof key
+  // If something changes in columns, you just need to update sortableKeys
+  // with the corresponding type
+  // The data model of the sortable keys is:
+  // name: string
+  // originalValue: number
+  // newValue: number
+  // deltaPercentage: number
+  // confidence: number
+  // originalRuns: array
+  // newRuns: array
+  // default: string
+  sortBy = (columnName, data) => {
+    let { sortOrder, sortClass } = this.state;
+    const { sortedTestName, sortedColumnName } = this.state;
+    const { testName } = this.props;
+    const sortableKeys = {
+      name: 'string',
+      originalValue: 'number',
+      originalStddevPct: 'number',
+      newValue: 'number',
+      newStddevPct: 'number',
+      deltaPercentage: 'number',
+      confidence: 'number',
+      originalRuns: 'array',
+      newRuns: 'array',
+    };
+    const sorted = true;
+    if (sorted) {
+      // ^ sorted flag becomes true after clicking for the first time any column
+      // Toggle sort order if clicked a column header several consecutive times
+      if (sortedTestName === testName && sortedColumnName === columnName) {
+        data.sort(sortHashes(columnName, sortableKeys[columnName], !sortOrder));
+      } else {
+        // sort ascending if any column was clicked for the first time
+        sortOrder = true;
+        data.sort(sortHashes(columnName, sortableKeys[columnName], !sortOrder));
+      }
+      if (sortOrder) sortClass = 'sorted-asc ';
+      else sortClass = 'sorted-desc ';
+    } else sortClass = '';
```

Comment:
There's a lot of nested if/else statements here. Can you give `sortClass` a default value and only change it if a condition is met?

Reply:
Default value of `sortClass` is `''`. If the column is sorted ascending, then the header will have bottom border 1px `'sorted-asc'`, if descending then top border 1px `'sorted-desc'`. We don't want any border to be visible while any column wasn't sorted (at page load) so default class is `''` (no class). 
What's complicating the `ifs` is that if we sort sequentially first column, then second, then third, etc. we want the sortOrder flag to be reset (to `true` -> `'sorted-asc'`) otherwise it will sort first column ascending, second descending and so on. So UX reasons.
I can't see right now a simpler and efficient way of doing it, but if you have suggestions I'm open.

Reply:
Ok, sure - I see what you're saying. But these are the things I'm finding confusing about this code:
-  You're extracting `sortClass` from state, but you're not actually using that value; you're only reassigning it and then updating state (setState). 

- sorted is always true - since you're assigning it to true - so if that's the case why do you need the if/else statement with `sortClass = '';` line?

- using the brackets with the else statement makes it easier to see what's in that block and what isn't